### PR TITLE
Storage tests

### PR DESCRIPTION
**Pull request by RaoulSchaffranek**
See #1307

### CODE REVIEW

File: tests/gen-spec/foundry/foundry.toml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[profile.default]
+src = 'src'
+out = 'out'
+rpc_endpoints = { optimism = "https://optimism.alchemyapi.io/v2/...", mainnet = "${RPC_MAINNET}" }
+
```
Comment (ehildenb) [inline tests/gen-spec/foundry/foundry.toml:5]: This whole directory goes away.

**Comment by ehildenb** [comment]
Can you back out Lisandra's changes, and only keep the `tests/specs/examples/Storage.sol` file and `tests/specs/examples/storage-spec.md`.

File: tests/specs/examples/Storage.sol [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    string public myString;
+    bytes public myBytes;
+    constructor () {
+        myBool = true;
+        myUint128 = 100;
+        myUint256 = 200;
+        myString = "foo";
+    }
+
+}
```
Comment (ehildenb) [inline tests/specs/examples/Storage.sol:82]: This looks good too, probably already this is revealing some storage slot types we don't support. Can we rebase this on master and get a clean diff?

**Comment by ehildenb** [comment]
We should also add nested storage types and such, for example a `Foo[]`, here `Foo` is a `struct`. Or a `struct` member which is a mapping of arrays of other struct members.

**Comment by RaoulSchaffranek** [comment]
New PR: #1319

**Pull request closed without merging**